data/lua/cave_map_generator.lua

```diff
@@ -0,0 +1,205 @@
+local helper = wesnoth.require "lua/helper.lua"
+local T = helper.set_wml_tag_metatable {}
+local random = wesnoth.random
+
+local params = _G.params
+
+local map = {}
+
+local function loc_to_index(x,y)
+	return x + 1 + y * params.map_width
+end
+
+local function index_to_loc(index)
+	return (index -1) % params.map_width, math.floor((index -1) / params.map_width)
+end
```

Comment:
These two functions have the look of something that should be offered directly in the Lua API, as all generators will require them. Though their current implementation isn't exactly general enough...